clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4012,7 +4012,7 @@ LangAS CodeGenModule::getStringLiteralAddressSpace() const {
     //   const char *getLiteral() n{
     //     return "AB";
     //   }
-    return LangAS::opencl_private;
+    return LangAS::opencl_global;
```

Comment:
To me, `opencl_constant` looks like a right value to return. Could you explain why we use `opencl_global` here?

Reply:
There are too many constraints on OpenCL `constant` address spaces: limited capacity, pointers that cannot be cast to `generic` address space...
So even it seems like the obvious answer, it causes a lot of trouble.
But since you are asking: there is obviously a missing comment explaining the motivation in the code... :-(
Actually any question in a PR means at least 1 lacking comment in the code. ;-)

Reply:
In this case though, there is a comment just above the differences that says:
    ```
    // If we keep a literal string in constant address space, the following code
    // becomes illegal:
    //
    //   const char *getLiteral() n{
    //     return "AB";
    //   }
```

Reply:
Yes but it does not explains why it becomes illegal.
The more comments, the happier I am. :-)

Reply:
There is a addressspacecast to generic address space in IR, but adressspacecast from constant to generic forbitten because of constant address space is not part of generic address space. @keryell, should I write it in source code comments?

Reply:
Yes please, add a lot of comments useful to understand the big picture. Thanks.